process/block/preprocess/sovereignRewardTxPreProcessor.go

```diff
@@ -0,0 +1,90 @@
+package preprocess
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/hashing"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/sharding"
+	"github.com/multiversx/mx-chain-go/state"
+)
+
+type sovereignRewardsTxPreProcessor struct {
+	*rewardTxPreprocessor
+}
+
+// ArgsRewardTxPreProcessor is a structure placeholder for arguments needed to create a rewards tx pre-processor
+type ArgsRewardTxPreProcessor struct {
+	RewardTxDataPool           dataRetriever.ShardedDataCacherNotifier
+	Store                      dataRetriever.StorageService
+	Hasher                     hashing.Hasher
+	Marshalizer                marshal.Marshalizer
+	RewardProcessor            process.RewardTransactionProcessor
+	ShardCoordinator           sharding.Coordinator
+	Accounts                   state.AccountsAdapter
+	OnRequestRewardTransaction func(shardID uint32, txHashes [][]byte)
+	GasHandler                 process.GasHandler
+	PubkeyConverter            core.PubkeyConverter
+	BlockSizeComputation       BlockSizeComputationHandler
+	BalanceComputation         BalanceComputationHandler
+	ProcessedMiniBlocksTracker process.ProcessedMiniBlocksTracker
+	TxExecutionOrderHandler    common.TxExecutionOrderHandler
+}
+
+// NewSovereignRewardsTxPreProcessor creates a sovereign rewards tx pre-processor
+func NewSovereignRewardsTxPreProcessor(args ArgsRewardTxPreProcessor) (*sovereignRewardsTxPreProcessor, error) {
```

Comment:
We do not need SovereignRewardsTXPreProcessor for the SovereignChain - on the mainchain we do not have this in MetaChain.

Reply:
Right, deleted it and now we don't have any more rewards pre processor added for sovereign